syntax = "proto3";

option go_package = "escher/private/grpc";

package escher;

import "escher.proto";

//Escher server from Payload side
service EscherPayload {
  rpc EscherCommunication(EscherMessage) returns (EscherMessageResponse) {}
  rpc Start(PayloadStartMessage) returns (PayloadStartResponse) {}
  rpc Terminate(PayloadTerminateMessage) returns (PayloadTerminateResponse) {}
}

//Escher Payload server from Escherlet side
service EscherletPayload {
  rpc PayloadReady(PayloadReadyMessage) returns (PayloadReadyResponse) {}
  rpc PayloadInactive(PayloadInactiveMessage) returns (PayloadInactiveResponse) {}
  rpc EscherCommunication(EscherMessage) returns (EscherMessageResponse) {}
}

message PayloadReadyMessage {
  string info = 1;
  uint32 port = 2;
  escher.EscherVersion version = 3;
}

message PayloadReadyResponse {
  string info = 1;
}

message PayloadInactiveMessage {
  string info = 1;
  uint64 since = 2;
}

message PayloadInactiveResponse {
  string info = 1;
}

message PayloadStartMessage {
  string info = 1;
}

message PayloadStartResponse {
  string info = 1;
}

message PayloadTerminateMessage {
  string info = 1;
}

message PayloadTerminateResponse {
}
